A fluid-type post

mirisuzanne opened the pull request:
![](https://source.unsplash.com/featured/?cute,animal&water)
<!-- Swap `CHANGE_ME` for a random string to get a different image -->


## Related Issue(s)
_Reminder to add related issue(s), if available._

- #530 

## Steps to test/reproduce
_Please explain how to best reproduce the issue and/or test the changes locally (including the pages/URLs/views/states to review)._


## Show me
_Provide screenshots/animated gifs/videos if necessary._

dvdherron commented on content/blog/2025/fluid-type.md:
```diff
@@ -0,0 +1,356 @@
+and set our base font size to `1.5em`.
+_Good enough_.
+
+TL;DR never do pixel math with `em` and `rem` units.
```
```suggestion
TL;DR: never do pixel math with `em` and `rem` units.
```

dvdherron commented on content/blog/2025/fluid-type.md:
```diff
@@ -0,0 +1,356 @@
+But I still have an issue
+because _the browser_ doesn't give me many options
+for setting the preference.
+And (not surprising)
```
```suggestion
And (not surprisingly)
```

dvdherron commented on content/blog/2025/fluid-type.md:
```diff
@@ -0,0 +1,356 @@
+Of course,
+I could set different defaults
+for different devices,
+But that doesn't help when I resize windows.
```
```suggestion
but that doesn't help when I resize windows.
```

dvdherron commented on content/blog/2025/fluid-type.md:
```diff
@@ -0,0 +1,356 @@
+Even with the browser constantly running fullscreen,
+may laptop is only sometimes plugged into
+an external monitor.
+when I unplug it, I have less space available.
```
```suggestion
When I unplug it, I have less space available.
```

dvdherron commented on content/blog/2025/fluid-type.md:
```diff
@@ -0,0 +1,356 @@
+We still want to make sure the defaults are reasonable,
+based on the global preferences,
+and only provide the UI as a progressive enhancement.
+To quote accessibility expert Kate Kalcevich…
```
```suggestion
To quote accessibility expert Kate Kalcevich:
```

dvdherron reviewed (commented):
Some copy feedback on a first pass.

stacyk commented on content/blog/2025/fluid-type.md:
````diff
@@ -0,0 +1,356 @@
+
+```css
+:root {
+  font-size: clamp(1.125rem, 1.0815rem + 0.2174vi, 1.25rem)
````
```suggestion
  font-size: clamp(1.125rem, 1.0815rem + 0.2174vi, 1.25rem);
```

stacyk commented on content/blog/2025/fluid-type.md:
```diff
@@ -0,0 +1,356 @@
+Now that I've said I _want_ `24px` font-size,
+my site has increased to `36px`!
+
+It seems like Chromium browsers
+no longer allow setting a specific font size.
+Instead you can choose from the default `medium`
+or a list of larger and smaller options:
+
+- Very small
+- Small
+- Medium (recommended)
+- Large
+- Very large
```
It does show the size when you go into the Customize Fonts panel and focus or hover over the Font Size slider
![font-size-slider](https://github.com/user-attachments/assets/6af4194b-0fc6-470b-9629-71df311099e2)

mirisuzanne replied:
Ah, that's interesting. I guess I hadn't gone deeper.

stacyk commented on content/blog/2025/fluid-type.md:
Should we add a link here so they know where to let us know what they think?

stacyk commented on content/blog/2025/fluid-type.md:
```diff
@@ -0,0 +1,356 @@
+But I've already designed my site
+to have large text _relative to_ the user default.
+Ignoring the fluid bits for now,
+I set the root `font-size` to roughly `1.5em` --
+half again the size of the user setting.
+When my preference was set to the default `16px`,
+my site would already display text at `24px`.
+Now that I've said I _want_ `24px` font-size,
+my site has increased to `36px`!
```
I don't think it matters that isn't what you are using as you could change that at any time after publishing this.

mirisuzanne commented on content/blog/2025/fluid-type.md:
```diff
@@ -0,0 +1,356 @@
+Again I've heard a common refrain that
+"users don't resize their windows" --
+but I find that entirely unbelievable.
+In fact, I'd bet I'm in a very small minority
+by using mostly fullscreen apps.
```
We could link to a mastodon or bluesky survey, but we'd be getting response from other developers. I guess the framing could be something like:

Do you ever resize your browser window, aside from testing responsive designs as a developer?

- Yes (I'm a front-end dev)
- Yes (Not a front-end dev)
- No (I'm a front-end dev)
- No (Not a front-end dev)

stacyk replied:
I think if you emphasize the "**aside from responsive testing**" then you might not need the "I'm a FED/not a FED" but that's my two cents.

mirisuzanne replied:
It looks like Bluesky doesn't have polls - but I set this one on Mastodon to run for three days: https://front-end.social/@mia/113992409837989769

mirisuzanne commented:
@stacyk I think I addressed your comments. @jgerigmeyer do you want to take a look as well? Ideally I'll post this today, since I'm out of the office tomorrow.

jgerigmeyer commented on content/blog/2025/fluid-type.md:
```diff
@@ -0,0 +1,404 @@
+{% callout 'note' %}
+We talked about some of these ideas
+in our [Winging It](/2025/01/24/winging-it-15/) live stream
+last month,
```
```suggestion
last month
```

jgerigmeyer commented on content/blog/2025/fluid-type.md:
```diff
@@ -0,0 +1,404 @@
+We talked about some of these ideas
+in our [Winging It](/2025/01/24/winging-it-15/) live stream
+last month,
+with Alan Stearns --
```
```suggestion
with Alan Stearns,
```

jgerigmeyer commented on content/blog/2025/fluid-type.md:
```diff
@@ -0,0 +1,404 @@
+Utopia asks us to start from a range of font sizes
+defined in `px` values,
+and then it does a conversion to `rem`
+by assuming that `1rem === 16px`.
```
This `===` feels very JavaScripty to me.

```suggestion
by assuming that `1rem == 16px`.
```

jgerigmeyer commented on content/blog/2025/fluid-type.md:
```diff
@@ -0,0 +1,404 @@
+defined in `px` values,
+and then it does a conversion to `rem`
+by assuming that `1rem === 16px`.
+As long as that assumption hold true,
```
```suggestion
As long as that assumption holds true,
```

jgerigmeyer commented on content/blog/2025/fluid-type.md:
```diff
@@ -0,0 +1,404 @@
+but sites with the same size or larger text
+certainly shouldn't get _even bigger_.
+
+It's not just my site,
```
```suggestion
It's not just my site --
```

jgerigmeyer commented on content/blog/2025/fluid-type.md:
```diff
@@ -0,0 +1,404 @@
+
+I see this situation play out over and over
+on the web.
+And then the lesson we often learn is
```
```suggestion
The lesson we often learn is
```

jgerigmeyer commented on content/blog/2025/fluid-type.md:
```diff
@@ -0,0 +1,404 @@
+and set our base font size to `1.5em`.
+_Good enough_.
+
+TL;DR: never do pixel math with `em` and `rem` units.
```
Elsewhere in OddSite we're doing:

```suggestion
TL;DR -- Never do pixel math with `em` and `rem` units.
```

jgerigmeyer commented on content/blog/2025/fluid-type.md:
```diff
@@ -0,0 +1,404 @@
+
+TL;DR: never do pixel math with `em` and `rem` units.
+That's where we went wrong,
+by assuming that `16px === 1em` is a reliable fact.
```
```suggestion
by assuming that `16px == 1em` is a reliable fact.
```

jgerigmeyer commented on content/blog/2025/fluid-type.md:
```diff
@@ -0,0 +1,404 @@
+
+## Why should type be responsive at all?
+
+But I still have an issue
```
```suggestion
But I still have an issue,
```

jgerigmeyer commented on content/blog/2025/fluid-type.md:
```diff
@@ -0,0 +1,404 @@
+for different devices,
+but that doesn't help when I resize windows.
+Even with the browser constantly running fullscreen,
+may laptop is only sometimes plugged into
```
```suggestion
my laptop is only sometimes plugged into
```

jgerigmeyer commented on content/blog/2025/fluid-type.md:
```diff
@@ -0,0 +1,404 @@
+Relative units like `em` and `rem`
+are excellent for _internal relationships_ --
+keeping the scale consistent between
+body text and headings for example.
```
```suggestion
body text and headings, for example.
```

jgerigmeyer commented on content/blog/2025/fluid-type.md:
```diff
@@ -0,0 +1,404 @@
+I think browsers like Arc are on the right track
+with per-site style adjustments.
+
+We could take my responsive-default approach even farther
```
```suggestion
We could take my responsive-default approach even further
```

jgerigmeyer commented on content/blog/2025/fluid-type.md:
```diff
@@ -0,0 +1,404 @@
+We still want to make sure the defaults are reasonable,
+based on the global preferences,
+and only provide the UI as a progressive enhancement.
+To quote accessibility expert Kate Kalcevich:
+
+{{ quotes.blockquote({
+  'text': "The real advance in accessibility is providing options and adapting to user preferences.",
+  'name': "Kate Kalcevich"
+}) }}
```
Can we link to a source for this quote? Not a big deal.

mirisuzanne replied:
It was a talk that I saw in person, so not sure if there's a link

mirisuzanne replied:
Found it, link added

jgerigmeyer commented on content/blog/2025/fluid-type.md:
```suggestion
I'm sure there are ways to improve it,
```

jgerigmeyer commented on content/blog/2025/fluid-type.md:
This codepen has two misspellings of "calculate"/"calculator" as "calcuate"/"calcuator".

mirisuzanne replied:
This should be fixed
